File: 백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int m; cin >> m;
    vector<int> p(m);
    for(int i = 0; i < m; i++) cin >> p[i];
    int pre = p[0], pcnt, ncnt, ans; pcnt = ncnt = ans = 1;
    for(int i = 1; i < m; i++){
        if(pre == p[i]) continue;
        if(pre > p[i]){
            ncnt++; pcnt = 1;
        }
        else{
            pcnt++; ncnt = 1;
        }
        pre = p[i];
        ans = max({ans, pcnt, ncnt});
    }
    cout << ans;
    return 0;
} 

File: 백준/Silver/27972. 악보는 거들 뿐/악보는 거들 뿐.py
import sys 
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)
input = lambda: sys.stdin.readline().rstrip('\r\n')

def main():
    m = int(input())
    p = list(map(int, input().split()))
    pre = p[0]
    pcnt = ncnt = ans = 1
    for i in range(1, m):
        if p[i-1] == p[i]: continue
        if p[i-1] > p[i]: ncnt += 1; pcnt = 1
        else: pcnt += 1; ncnt = 1
        ans = max(ans, pcnt, ncnt)       
    print(ans)
    return
if __name__ == '__main__':
    main()